packages/trader/src/Modules/SmartChart/Components/AllMarkers/Helpers/calculations.js

```diff
@@ -0,0 +1,14 @@
+export const calc_scale = zoom => {
+    return zoom ? Math.max(Math.min(Math.sqrt(zoom / 18), 1.2), 0.8) : 1;
+};
+
+const hex_map = [];
+export const calc_opacity = (from, to) => {
+    if (hex_map.length === 0) {
```

Comment:
can this if statement be moved outside? it seems to be filling the hex_map array with hex values, which i think needs to be done only once

Reply:
It does, however it only do it once when the hex_map is empty.

Reply:
ah I see 👌